graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -165,32 +143,96 @@ def class_fields(self):
     @property
     def is_vertex(self):
         """Return True if the schema element represents a vertex type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_VERTEX
+        return isinstance(self, VertexType)
 
     @property
     def is_edge(self):
         """Return True if the schema element represents an edge type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_EDGE
+        return isinstance(self, EdgeType)
 
     @property
     def is_non_graph(self):
         """Return True if the schema element represents a non-graph type, and False otherwise."""
-        return self._kind == SchemaElement.ELEMENT_KIND_NON_GRAPH
+        return isinstance(self, NonGraphElement)
+
+    def freeze(self):
+        """Do nothing."""
```

Comment:
Explain what `freeze` is supposed to do. The `pass` makes it obvious that it doesn't happen here, but you should explain to implementers of subclasses what this method is intended to do, so they know when to override it with a custom implementation.

Reply:
Fixed in "nits" pr.